lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/RestAccessControlRulesProvider.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.security;
+
+import io.airlift.http.client.HttpClient;
+import io.airlift.http.client.HttpStatus;
+import io.airlift.http.client.Request;
+import io.airlift.http.client.StringResponseHandler.StringResponse;
+
+import java.net.URI;
+import java.util.function.Supplier;
+
+import static io.airlift.http.client.Request.Builder.prepareGet;
+import static io.airlift.http.client.StringResponseHandler.createStringResponseHandler;
+import static io.trino.plugin.base.security.FileBasedAccessControlUtils.parseJSONString;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class RestAccessControlRulesProvider<R>
```

Comment:
This class looks very similar to `LocalFileAccessControlRulesProvider`, can you please extract common parts?